fabric-network/test/impl/event/contractlistener.spec.ts

```diff
@@ -87,76 +87,74 @@ describe('contract event listener', () => {
 		});
 	}
 
-	function newFullBlock(): Block {
-		const block = new protos.common.Block();
-		block.data = new protos.common.BlockData();
-		block.metadata = new protos.common.BlockMetadata();
+	function newFullBlock(): fabproto6.common.Block {
+		const block = new fabproto6.common.Block();
+		block.data = new fabproto6.common.BlockData();
+		block.metadata = new fabproto6.common.BlockMetadata();
 		block.metadata.metadata = [];
-		block.metadata.metadata[protos.common.BlockMetadataIndex.TRANSACTIONS_FILTER] = [];
+		block.metadata.metadata[fabproto6.common.BlockMetadataIndex.TRANSACTIONS_FILTER] = new Uint8Array(10);
 		return block;
 	}
 
-	function addTransaction(event: EventInfo, transaction: any, statusCode: number = protos.protos.TxValidationCode.VALID, transactionId?: string): void {
+	function addTransaction(event: any, transaction: any, statusCode: number = fabproto6.protos.TxValidationCode.VALID, index: number = 0, transactionId?: string): void {
 		event.block.data.data.push(newEnvelope(transaction, transactionId));
-
-		const transactionStatusCodes = event.block.metadata.metadata[protos.common.BlockMetadataIndex.TRANSACTIONS_FILTER];
-		transactionStatusCodes.push(statusCode);
+		event.block.metadata.metadata[fabproto6.common.BlockMetadataIndex.TRANSACTIONS_FILTER][index] = statusCode;
 	}
 
 	function newEnvelope(transaction: any, transactionId?: string): any {
-		const channelHeader = new protos.common.ChannelHeader();
-		channelHeader.type = protos.common.HeaderType.ENDORSER_TRANSACTION;
+		const channelHeader = new fabproto6.common.ChannelHeader();
+		channelHeader.type = fabproto6.common.HeaderType.ENDORSER_TRANSACTION;
 		channelHeader.tx_id = transactionId;
 
-		const payload = new protos.common.Payload();
-		payload.header = new protos.common.Header();
+		const payload: any = {};
```

Comment:
Might be safer to stick with the protobuf definitions rather than `any` and just cast values we're assigning to properties to `any` type at assignment where the object representation doesn't match the protobuf TypeScript. At least the typing will ensure we get the correct property names that way

Reply:
done